molgenis-security/src/main/java/org/molgenis/security/account/AccountServiceImpl.java

```diff
@@ -158,14 +158,24 @@ private String createActivationEmailText(User user, URI activationUri) {
   }
 
   private String createActivatedEmailText(User user, String appName) {
-    return "Dear "
-        + user.getFirstName()
-        + " "
-        + user.getLastName()
-        + ",\n\n"
-        + "your registration request for "
-        + appName
-        + " was approved.\n"
-        + "Your account is now active.\n";
+    String firstName = user.getFirstName();
+    String lastName = user.getLastName();
+    if (firstName == null || firstName.isEmpty()) firstName = user.getUsername();
+    if (lastName == null || lastName.isEmpty()) lastName = null;
+    StringBuilder builder = new StringBuilder();
+    builder.append("Dear ");
+    builder.append(firstName);
+    if (lastName != null) {
+      builder.append(" ");
+      builder.append(lastName);
+    }
+    builder.append(",\n\n");
+    builder.append("Your registration request for ");
+    builder.append(appName);
+    builder.append(" was approved.\n");
+    builder.append("Your account is now active.\n\n");
+    builder.append("Kind regards,\n\n");
+    builder.append("MOLGENIS support");
```

Comment:
"MOLGENIS support" might be a bit strange for third party's hosting a molgenis instance...